mobile/lib/providers/likes_providers.dart

```diff
@@ -62,13 +75,21 @@ class LikesNotifier extends _$LikesNotifier {
   ///
   /// Called when user authenticates. Syncs with local storage and relays.
   Future<void> _initialize() async {
+    // Create a new completer for this initialization
+    _initializationCompleter = Completer<void>();
+
+    // Small delay to allow likesRepositoryProvider to rebuild after invalidation
+    await Future<void>.delayed(const Duration(milliseconds: 50));
```

Comment:
This seem to me like hack not solution with the Completer, isnt there better solution?